addons/mail/static/src/utils/common/local_storage.js

```diff
@@ -0,0 +1,62 @@
+import { browser } from "@web/core/browser/browser";
+import { session } from "@web/session";
+
+const LOCAL_STORAGE_SUBVERSION = 0;
+
+/**
+ * @typedef {Object} VersionedValue
+ * @property {any} value
+ * @property {string} version
+ */
+
+export function getCurrentLocalStorageVersion() {
+    const [major, minor] = session.server_version_info ?? "1.0";
+    return [major, minor, LOCAL_STORAGE_SUBVERSION].join(".");
+}
+
+/**
+ * Utility class to simplify interaction on local storage with constant local storage key and with versioning.
+ * When a value is set, this is done as `{ value, version }`.
```

Comment:
We might want to comment the object syntax is necessary regardless of the version to properly handle types ("false" vs false, ...)